internal/sync/lifecycle.go

```diff
@@ -83,23 +121,25 @@ func (l *LifecycleOnce) Start(f func() error) error {
 	// called the start function already, or called the stop function
 	// in which case we should exit now and return the result of the
 	// last start command (or nil).
-	if LifecycleState(l.state.Load()) != Idle {
+	if state := LifecycleState(l.state.Load()); state != Idle {
```

Comment:
not necessary, but helps readability to separate the coercion from the comparison.